website/static/js/pages/project-settings-page.js

```diff
@@ -54,8 +54,73 @@ if ($('#wgrid').length) {
         });
     });
 }
+
+
+var institutionsViewModel = function() {
+    ko.punches.enableAll();
+    var self = this;
+    self.primaryInstitution = ko.observable('None');
+    self.institutionHref = ko.observable('');
+    self.availableInstitutions = ko.observable();
+    $osf.ajaxJSON(
```

Comment:
Avoid making requests in constructors; they make classes hard to test.

```js
// I want to unit test InstitutionsViewModel, let's instantiate one
var vm = new InstitutionsViewModel();
// Ah! it's sending an API request. Now I need to mock stuff... ='(
```

Better to separate fetching into methods.

```js
var vm = new InstitutionsViewModel();
vm.fetchUserInstitutions(userID)
    .done(function(institutions) { console.log(institutions); });
```